test/data/hello.sh

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```

Comment:
Could this be OS agnostic?
Windows also has scripts (and `echo` is also a Windows command`).

Or, there could be another script (like `hello.bat`) that is then tested under Windows.

Reply:
I will have to think about this. I think that a `hello.bat` file makes sense, although I am not sure if windows support dots like in unix. Something I have to point out is that (I think?) currently Shelly cannot execute things without the dot unless you use `escaping False`, and this is independent from the `system-filepath` regression. To be precise, I think something like this fails in Shelly:

```hs
      it "script (without dot) at $PWD" $ do
        res <- shelly $ do
          run_ "chmod" ["+x", "test/data/hello.sh"]
          run "test/data/hello.sh" []
        res @?= "Hello!\n"
```

I think that's because that `whichEith` function will consider this _not_ an absolute path (it's relative) and it doesn't have dot in it, so it will try to find the whole thing at `PATH` and things will fail. But I will suggest not trying to patch this in this PR, to not inflate its size.

Reply:
Ok, we do not have to improve on Shelly in this PR, just restore its old behavior...